test/integration/good/code-gen/cpp.expected

```diff
@@ -20110,6 +20110,573 @@ stan::model::model_base& new_model(
+          context__.to_vec());
+      nt = std::numeric_limits<int>::min();
+      
+      current_statement__ = 3;
+      nt = context__.vals_i("nt")[(1 - 1)];
+      current_statement__ = 4;
+      context__.validate_dims("data initialization","NS","int",
+          context__.to_vec());
+      NS = std::numeric_limits<int>::min();
+      
+      current_statement__ = 4;
+      NS = context__.vals_i("NS")[(1 - 1)];
+      current_statement__ = 5;
+      validate_non_negative_index("L_Omega", "nt", nt);
+      current_statement__ = 6;
+      validate_non_negative_index("z1", "NS", NS);
+    } catch (const std::exception& e) {
+      stan::lang::rethrow_located(e, locations_array__[current_statement__]);
+      // Next line prevents compiler griping about no return
+      throw std::runtime_error("*** IF YOU SEE THIS, PLEASE REPORT A BUG ***"); 
+    }
+    num_params_r__ = 0U;
+    
+    try {
+      num_params_r__ += nt * ((2 * (2 - 1)) / 2);
+      num_params_r__ += NS;
+    } catch (const std::exception& e) {
+      stan::lang::rethrow_located(e, locations_array__[current_statement__]);
+      // Next line prevents compiler griping about no return
+      throw std::runtime_error("*** IF YOU SEE THIS, PLEASE REPORT A BUG ***"); 
+    }
+  }
+  template <bool propto__, bool jacobian__, typename VecR, typename VecI, stan::require_vector_like_t<VecR>* = nullptr, stan::require_vector_like_vt<std::is_integral, VecI>* = nullptr>
+  inline stan::scalar_type_t<VecR> log_prob_impl(VecR& params_r__,
+                                                 VecI& params_i__,
+                                                 std::ostream* pstream__ = nullptr) const {
+    using T__ = stan::scalar_type_t<VecR>;
+    using local_scalar_t__ = T__;
+    T__ lp__(0.0);
+    stan::math::accumulator<T__> lp_accum__;
+    static const char* function__ = "param_constraint_model_namespace::log_prob";
+(void) function__;  // suppress unused var warning
+
+    stan::io::reader<local_scalar_t__> in__(params_r__, params_i__);
+    local_scalar_t__ DUMMY_VAR__(std::numeric_limits<double>::quiet_NaN());
+    (void) DUMMY_VAR__;  // suppress unused var warning
+
+    
+    try {
+      std::vector<Eigen::Matrix<local_scalar_t__, -1, -1>> L_Omega;
+      L_Omega = std::vector<Eigen::Matrix<local_scalar_t__, -1, -1>>(nt, Eigen::Matrix<local_scalar_t__, -1, -1>(2, 2));
+      stan::math::fill(L_Omega, DUMMY_VAR__);
+      
+      std::vector<Eigen::Matrix<local_scalar_t__, -1, 1>> L_Omega_in__;
+      L_Omega_in__ = std::vector<Eigen::Matrix<local_scalar_t__, -1, 1>>(nt, Eigen::Matrix<local_scalar_t__, -1, 1>(
+        ((2 * (2 - 1)) / 2)));
+      stan::math::fill(L_Omega_in__, DUMMY_VAR__);
+      
+      current_statement__ = 1;
+      for (int sym1__ = 1; sym1__ <= nt; ++sym1__) {
+        current_statement__ = 1;
+        assign(L_Omega_in__, cons_list(index_uni(sym1__), nil_index_list()),
+          in__.vector(((2 * (2 - 1)) / 2)), "assigning variable L_Omega_in__");
+      }
```

Comment:
Will this run? Reader has state so idt you can call `in__.vector(N)` multiple times just getting out pieces like this.

 `in__` here is a [`reader`](https://github.com/stan-dev/stan/blob/develop/src/stan/io/reader.hpp#L189) class  which stores a reference to the input data as a serialized vector. When we call `in__.vector(N)` you can see in the link above we grab a block of data from the serialized vector of size N then increment an internal sentinel value so the next call will grab the next chunk of data.

Can this just be generated as

```suggestion
current_statement__ = 1;
assign(L_Omega_in__, nil_index_list(),
  in__.vector(((2 * (2 - 1)) / 2)), "assigning variable L_Omega_in__");
```

Reply:
actually nvm about this comment. I misread the code and this is reading in vectors of size `((2 * (2 - 1)) / 2)` `nt` times which is the right behavior